cmd/query/app/flags.go

```diff
@@ -46,11 +46,13 @@ const (
 	queryTokenPropagation   = "query.bearer-token-propagation"
 	queryAdditionalHeaders  = "query.additional-headers"
 	queryMaxClockSkewAdjust = "query.max-clock-skew-adjustment"
+	queryHTTPTLSEnabled     = "query.http.tls.enabled"
+	queryGRPCTLSEnabled     = "query.gprc.tls.enabled"
```

Comment:
why do we need these? `enabled` is already a standard flag in the TLS config https://github.com/jaegertracing/jaeger/blob/master/pkg/config/tlscfg/flags.go#L26

Reply:
We need two one for GRPC one for HTTP.  but even if either one of them are enabled, we need only one set of certificates and keys.  So,  I used two hardCoded flags for each and used `showEnabled:false ` for the common certificate configuration.

Reply:
using enabled of `tlsCfg` would also force two sets of certificate and key.  https://github.com/jaegertracing/jaeger/issues/2338#issuecomment-659955372

Reply:
I don't follow. If you want to share certificates and keys between two servers, then shouldn't we have only one `TLS tlscfg.Options` field in the options struct, as we had before? Then it would make sense to have to extra `.enabled` flags to allow switching TLS on each server separately.

Reply:
Sure. I'll do that.

Does the project have any convention for multi-vlaued flags?  Because clientCA of GRPC and HTTP server might be different.  Currently we still have this problem.

Reply:
The user must concatenate all certificates int  a single certificate file.  I guess it would work. Thanks.

Reply:
FWIW, I'd say that, atleast  when using mutual tls, it's unlikely you'll want the same server certs and CA for https and grpc. The grpc services will generally have certs issued from a common infrastructure CA controlled by the user, whereas user facing https will probably want a cert signed by a public CA, recognized by browsers.

Reply:
Then I think we would need two separate ports for the two different servers.  I am unaware how to use a single port with [cmux](https://godoc.org/github.com/soheilhy/cmux) with two different TLS certificates.

Reply:
I'd love to be proven wrong, but I don't think it's even possible. In order to determine the payload type, the headers have to be read, and this can only be done after the TLS handshake. Without knowing which cert to serve, there can't be a TLS handshake.

Reply:
I'm sure it's probably possible to use the regular TLS cert negotiation selection with cmux, but on the whole I'm really against the whole "do it all on one port" thing, it gets very confusing when debugging.
FWIW, SNI negotiation occurs before the TLS handshake, so you can at least know which cert the client wanted, but the client does have control over that, so it's all a bit dubious at that point.
We're using grpc mtls betweem agents in local clusters to a single remote collector.

Reply:
@jpkrohling  Agreed. So our options are:

1.  Either we have to use one certificate where the server uses only one port for both HTTP and gRPC endpoints.
1.  Use separate port for http and another port fro gRPC.  then,  each will have different listeners and we can use two set of certificates generated by two diffrent CAs.

Reply:
@tcolgate given that Jaeger 2.x will be based on OpenTelemetry Collector, you might want to bring your concerns to this issue as well: open-telemetry/opentelemetry-collector#1256. 

This PR here is about the query server, which has been released already with both protocols on the same port. Splitting it now would cause current clients to break, so, I'm afraid we don't have much choice at this point.

Reply:
@jpkrohling I did consider that. I added the mutual TLS support to jaeger and thanos, and then looked at doing it for otel, but cmux made it awkward, and debugging it did my brain in. (I disliked the cmux setup before that, but this was the first time I had to actually modify a project using it, that wasn't just ripping it out).
  Hopefully someone with more time an patience will do the otel work before I have to look at it again.

Reply:
@jpkrohling I've added a couple of comments to that bug. I hadn't realised they'd reverted the cmux usage. I failed to get mutual TLS working with cmux, I can't recall the exact details of why, but it looks like the bug that caused them to revert might be the reason.
  If the grpc service only supports functionality that is provided by the UI, then having the same CA/certs is probably reasonable, assuming none of the other jaeger components need grpc to talk to it (other than maybe a cli query, which would be roughly equivalent to browser usage, so could just use a system cert pack).

Reply:
@yurishkuro can confirm, but my understanding is that the query gRPC endpoint is (or is going to be) *the* supported way to query the backend. The first client is the UI, but external clients are also expected to exist, so, mTLS is certainly something we'd want to support there.